tests/graceful-recovery.md

```diff
@@ -0,0 +1,97 @@
+# Graceful recovery from restarts
+
+## Description
+When one of the containers in the NGF Pod fails, the Gateway API configuration
+should be reapplied to the new container so that when either container
+fails, NGF should be able to recover without any assistance.
+
+## Goal
+Ensure that NGF can recover gracefully from container failures without any intervention.
+
+## Cluster Details
+
+- GKE 1.27.3-gke.100
+- us-central1-c
+- Machine type of node is e2-medium
+- 3 nodes
+
+## Setup
+
+1. Setup GKE Cluster.
+2. Clone the repo and change into the nginx-gateway-fabric directory.
+3. Check out the latest tag (unless you are installing the edge version from the main branch).
+4. Go into `deploy/manifests/nginx-gateway.yaml` and change `runAsNonRoot` from `true` to `false`.
+5. Follow the [installation instructions](https://github.com/nginxinc/nginx-gateway-fabric/blob/main/docs/installation.md)
+to deploy NGINX Gateway Fabric.
+6. In a separate terminal track NGF logs by running `kubectl -n nginx-gateway logs -f deploy/nginx-gateway`
```

Comment:
I recommend using code block (three backticks) for any command that will be copy-pasted since GitHub will add a copy button to the snippet.